### PR TITLE
Add support for security labels

### PR DESCRIPTION
**Pull request by chrishajas**
Security labels were introduced in GPDB6 during the 9.1 merge.

* This also refactors some tests to use the DefaultMetadata and
DefaultMetadataMap functions instead of manually constructing the
metadata
* In some of the create tests, we aren't actually asserting that the
metadata is logically correct, but rather only syntactically correct. Since
the bulk of this code is the same, it's likely ok for now.

I still need to figure out how to get this to run in CI, but it's reviewable in this state.

Authored-by: Chris Hajas <chajas@pivotal.io>

### CODE REVIEW

File: backup/predata_acl.go [hunk 1]
```diff
@@ -403,6 +410,19 @@ func (obj ObjectMetadata) GetCommentStatement(objectName string, objectType stri
 	return commentStr
 }
 
+func (obj ObjectMetadata) GetSecurityLabelStatement(objectName string, objectType string) string {
+	providerStr := ""
+	securityLabelStr := ""
+	if obj.SecurityLabelProvider != "" {
```
Comment (khuddlefish) [inline backup/predata_acl.go]: I'm pretty sure all security labels have a provider. It is optional to specify one when creating then if you only have one provider loaded, but that just means the label gets that provider. We should probably always specify the provider for correctness.

File: backup/predata_acl.go [hunk 2]
```diff
@@ -91,6 +96,8 @@ func ConstructMetadataMap(results []MetadataQueryStruct) MetadataMap {
 			metadata.Privileges = make([]ACL, 0)
 			metadata.Owner = result.Owner
 			metadata.Comment = result.Comment
+			metadata.SecurityLabelProvider = result.SecurityLabelProvider
+			metadata.SecurityLabelTarget = result.SecurityLabelTarget
```
Comment (khuddlefish) [inline backup/predata_acl.go]: Is target the actual label? Is so, can we just call it "SecurityLabel"? label is what postgres seems to call it.

File: integration/predata_acl_queries_test.go [hunk 3]
```diff
@@ -511,7 +563,7 @@ LANGUAGE SQL`)
 				resultMetadataMap := backup.GetMetadataForObjectType(connectionPool, backup.TYPE_TYPE)
 
 				uniqueID := testutils.UniqueIDFromObjectName(connectionPool, "testschema", "testtype", backup.TYPE_TYPE)
-				expectedMetadata := testutils.DefaultMetadata("TYPE", false, true, true)
```
Comment (khuddlefish) [inline integration/predata_acl_queries_test.go]: I thought types can have security labels?

Comment (chrishajas) [reply integration/predata_acl_queries_test.go]: This is just testing that the logic works if the object is in a specific schema, there are already tests above that test it with security labels. The security label logic doesn't have any logic for schema filtering, so I just turned it off for those cases.

Comment (khuddlefish) [reply integration/predata_acl_queries_test.go]: Ok that's fine then

File: backup/queries_acl.go [hunk 4]
```diff
@@ -103,10 +103,12 @@ func InitializeMetadataParams(connectionPool *dbconn.DBConn) {
...
 	UniqueID
-	Privileges sql.NullString
-	Kind       string
-	Owner      string
-	Comment    string
+	Privileges            sql.NullString
+	Kind                  string
+	Owner                 string
+	Comment               string
+	SecurityLabelTarget   string
```
Comment (khuddlefish) [inline backup/queries_acl.go]: Remove "Target" here too

File: backup/queries_relations.go [hunk 5]
```diff
@@ -253,21 +253,23 @@ WHERE r.parchildrelid != 0;
...
+	StatTarget            int `db:"attstattarget"`
+	StorageType           string
+	DefaultVal            string
+	Comment               string
+	ACL                   []ACL
+	Options               string
+	FdwOptions            string
+	Collation             string
+	SecurityLabelProvider string
+	SecurityLabelTarget   string
```
Comment (khuddlefish) [inline backup/queries_relations.go]: Remove "Target" here too

**Comment by chrishajas** [comment]
@khuddlefish Renamed all instances of SecurityLabelTarget to SecurityLabel. Anything else?

**Pull request merged**